packages/stack/namesystem/src/index.js

```diff
@@ -43,6 +29,30 @@ export default class Namesystem {
     this.events.setCommandHandler("namesystem:registerSubdomain", (name, address, cb) => {
       this.executeNodeCommand('registerSubdomain', [name, address], cb);
     });
+
+    this.embark.events.setCommandHandler("module:namesystem:reset", (cb) => {
+      this.executeNodeCommand('reset', [], (_err) => {
+        // We ignore the error because it's just the reset, the node might have not been started at all
+        this.startNode(this.embark.config.namesystemConfig, cb);
+      });
+    });
+  }
+
+  startNode(namesystemConfig, cb) {
+    if (!namesystemConfig.enabled) {
+      return cb();
+    }
+    const nodeName = namesystemConfig.provider;
+    if (!nodeName) return cb(__("No namesystem provider specified in configs"));
+    const client = this.namesystemNodes[nodeName];
+    if (!client) return cb(__("Namesystem client %s not found", nodeName));
+
+    client.startFunction.apply(client, [
+      () => {
+        client.started = true;
+        cb();
+      }
+    ]);
```

Comment:
Can't we just do `client.startFunction(() => { ... })` here?

Reply:
Yup, most probably  (didn't write that code, but I should have tried)

Reply:
Updated